### PR TITLE
Commit options

### PR DESCRIPTION
**Pull request by imobachgs**

### CODE REVIEW

File: storage/Action.cc [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace storage
 
 
 	void
-	Create::commit(CommitData& commit_data) const
+	Create::commit(CommitData& commit_data, CommitOptions& commit_options) const
```
Comment (aschnell) [inline storage/Action.cc]: That parameter should be const.

File: storage/CommitOptions.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+#define STORAGE_COMMIT_OPTIONS_H
+
+namespace storage
+{
+    class CommitOptions
+    {
+    public:
+
+	//CommitOptions(const bool force_rw);
+	CommitOptions(const bool force_rw): force_rw(force_rw) {};
```
Comment (aschnell) [inline storage/CommitOptions.h]: The const here is not really required since the parameter is passed by value.

File: storage/CommitOptions.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+	~CommitOptions() {};
```
Comment (aschnell) [inline storage/CommitOptions.h]: The destructor should not be needed.

File: storage/Storage.h [hunk 4]
```diff
@@ -264,6 +264,13 @@ namespace storage
...
 	 * @throw Exception
 	 */
+	void commit(CommitOptions& commit_options, const CommitCallbacks* commit_callbacks = nullptr);
+
+	/**
+	 * The actiongraph must be valid.
+	 *
+	 * @throw Exception
+	 */
 	void commit(const CommitCallbacks* commit_callbacks = nullptr);
```
Comment (aschnell) [inline storage/Storage.h]: Should be marked as deprecated.

**Comment by aschnell** [comment]
LGTM

**Comment by imobachgs** [comment]
Thanks!

**Pull request merged**